### DISCUSSION TITLE
How to get ?xx= URL parameters in getServerSideProps

### DISCUSSION THREAD

**Discussion by derekang** [vercel/next.js#13309] [Help] [answered]
Hi, 

I have this URL at http://localhost:3000/e/women/dresses?color=black,blue

"dresses" is a dynamic [category]/js file. I am using getServerSideProps(), may I know how can get get the color GET parameter into getServerSideProps function?

I can get "dresses" with no problem using the below params, but I can't get color param.

```
export async function getServerSideProps({ params }) {
const category = params.category
...
}
```
Any guidance appreciated!

**Comment by jamesmosier** [comment, marked as answer]
```js
export async function getServerSideProps({ query }) {
  const category = query.category
  ...
}
```

You can grab the query string via `context.query`

**Reply by DeBraid** [reply]
Note above will NOT work inside `getStaticProps` as `query` is not defined as part of context.

Related feature request to add this: https://github.com/vercel/next.js/discussions/17269

**Reply by tNaratippakornDigio** [reply]
https://stackoverflow.com/a/61240697/14228603
````
export async function getServerSideProps(context) {
  const { id } = context.query;
  const res = await fetch(`https://restcountries.eu/rest/v2/name/${id}`);
  const country = await res.json();

  console.log(`Fetched place: ${country.name}`);
  return { props: { country } };
}

**Reply by agitrubard** [reply]
**For Use Both Query and Cookies Together in getServerSideProps**

```
export async function getServerSideProps(context) {
      const countryId = context.req.cookies['countryId'];
      const { id } = context.query;

      const res = await fetch(`https://restcountries.eu/rest/v2/name/${id}`);

      const country = await res.json();

      return { 
            props: {, 
                  countryNumber: countryId,
                  countryName: country.name
            } 
      }
}
```

**Comment by ryosuke-hujisawa** [comment]
```
export async function getServerSideProps(ctx: any) {

  const urlParam = ctx.query.id;
  console.log(urlParam);
```
[getServerSidePropsでURLパラメータ取得する方法](https://off.tokyo/blog/getserversideprops-url-2/)
